notes/pointer-basics.md

```diff
@@ -0,0 +1,219 @@
+# Pointers
+
+Contributors: Edward Zhang
+
+---
+A **pointer** is a type of variable that stores an address
+
+Pointers are most commonly used to indirectly access another object at some address
+- Thus, a pointer "points" to whatever data is located at the address it stores
+
```

Comment:
I'm not sure if this really matters, but I would be careful to reserve the word "object" for Objects, especially in the coding context. Maybe rename to variable. I'm noticing this throughout the paper

Reply:
Yeah I debated this for a while. Technically speaking, in C++ something like an `int` variable would be considered an object (see this [reference](https://en.cppreference.com/w/cpp/language/object)). 

On the other hand I think Java would say an `int` variable is a primitive type and reserve "object" status for actual instantiations of a class...

Any thoughts @markasoftware

Reply:
also this [SO post](https://stackoverflow.com/questions/48026922/is-an-int-variable-an-object-according-to-the-c-standard)

Reply:
Fair point, but perhaps we'll need to specify this somewhere so new recruits won't get confused